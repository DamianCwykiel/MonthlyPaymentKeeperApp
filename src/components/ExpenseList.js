import React from 'react';
import { connect } from 'react-redux';
import ExpeneListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//Adapter configuration
Enzyme.configure({adapter: new Adapter()});

export const ExpenseList = (props) => (
    <div className="content-container__add">
    <div className="list-header">
        <div className="see-on-mobile">Payments</div>
        <div className="see-on-desktop">Bill & Date</div>
        <div className="see-on-desktop">Amount</div>
    </div>
        <div className="list-body">
        {
        props.expenses.length === 0 ? (
            <div className="list-item">
                <span className="list-item__message">No payments to worry about!</span>
            </div>
        ) : (
            props.expenses.map((expense) => {
                return <ExpeneListItem key={expense.id} {...expense} />;
            })
        )
    }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        filters: state.filters
    }
};
export default connect(mapStateToProps)(ExpenseList); 

