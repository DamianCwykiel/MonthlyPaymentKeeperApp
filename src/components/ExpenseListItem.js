import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

//PLN currency
numeral.register('locale', 'pln', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: 'zł'
    }
});

// switch between locales
numeral.locale('pln');

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className="list-item" to = {`/edit/${id}`}>
        <div>
            <h3 className="list-item__title">{description}</h3>
            <h3 className="list-item__sub-title">{moment(createdAt).format('MMMM Do YYYY')}</h3>
        </div>
        <h3 className="list-item__data">
            <p>{numeral(amount / 100).format('0,0.00 $')}</p>
            <p>~ {numeral((amount / 100) / 4.30).format('0,0.00')} €.</p>
        </h3> 
    </Link>
); 
export default ExpenseListItem;