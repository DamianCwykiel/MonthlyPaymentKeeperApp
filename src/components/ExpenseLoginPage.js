import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const ExpenseLoginPage = ({ startLogin }) => (
<div className="box-layout">
    <div className ="box-layout__box">
        <h1 className="box-layout__title">Monthly$Payments€Keeper£App</h1>
        <p className="box-layout__subtitle">Use gmail account to log in to this application</p>
        <button className="login-button" onClick={startLogin}>Google login</button>
    </div>
</div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()) 
});

export default connect (undefined, mapDispatchToProps)(ExpenseLoginPage);