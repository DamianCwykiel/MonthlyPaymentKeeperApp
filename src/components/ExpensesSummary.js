import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expensesPhrase = expenseCount === 1 ? 'payment' : 'payments';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('0,0.00 $');
    const formattedExpenseTotalEuroCurrency = numeral ((expensesTotal / 100) / 4.29).format('0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h2 className="page-header__title">
                    Check your bills! There are: <span>{expenseCount}</span> {expensesPhrase}. In total that is: <span>{formattedExpensesTotal}</span>.
                    That equals ~ <span>{formattedExpenseTotalEuroCurrency} â‚¬</span>.
                    <div className="page-header__actions">
                        <Link className="create-button" to="/create">Add Your Bill</Link>
                    </div>
                </h2>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);