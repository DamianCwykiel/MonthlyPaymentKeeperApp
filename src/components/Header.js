import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Monthly$Payments€Keeper£App</h1>
                </Link>
                <button className="login-button button--link" onClick={startLogout}>Logout</button>
            </div>
            <div className="navlink-content">
                <Link className="header__about-app" to="/help">About App</Link>
            </div>
        </div>
    </header>
);
const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect (undefined, mapDispatchToProps)(Header);