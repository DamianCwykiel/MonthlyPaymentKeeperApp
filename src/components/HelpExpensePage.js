import React from 'react';
import {Link} from 'react-router-dom';

export const HelpExpensePage = () => (
    <div>
        <header className="header">
            <div className="content-container">
                <div className="header__content">
                    <h1 className="header__title">About App</h1>
                </div>
                <div className="navlink-content">
                    <Link className="header__about-app"to='/'>Home Page</Link>
                </div>
            </div>
        </header>
        <div className="page-header-about-app"> 
            <div className="content-container">
               <article className="list-item__message">
                    Monthly$Payments€Keeper£App was created as one of projects during WebDeveloper course.<br/>
                    This project uses including: React, Redux, .SCSS, Firebase, Webpack and was been tested in Jest.<br/>
                    The app styled as a home budget-app allows to add, change or remove created expenses. <br/>
                    The database was placed in the google Firebase and an access to it is via google account login.<br/>
                    In the app an amount in total you can see above of item's list in the header.<br/>
                    Addidtionally there you can check the currncy amount which was estimated according average 1 € ~ 4.30zł 
                    (March 2020).<br/>
                    <span>Caution!</span> At the first login don't interrupt the poping up auth window due to fail an authorisation process.<br/>
                    Due to prevent fail the app-page should be refresh or reopen.<br/>
                    In some cases if your browser after click on link below asks for certificate for this page you should click show certificate 
                    and add it to your browser certificate list. <br/>
                    If you want to see more projects like this click - <a href ="https://damiancwykiel-portfolio.pl" target="_blank">here.</a>
                </article>
            </div> 
        </div>  
    </div>
);
export default HelpExpensePage; 