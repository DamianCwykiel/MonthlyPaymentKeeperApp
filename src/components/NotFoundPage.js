import React from 'react';
import {Link} from 'react-router-dom';

export const NotFoundPage = () => (
    <div className="content-container__edit">
        <h4 className="header__title">
             <b>404! - Page not found.</b>
        </h4>
            <Link to='/'>Home Page</Link>
    </div>
);
export default NotFoundPage;