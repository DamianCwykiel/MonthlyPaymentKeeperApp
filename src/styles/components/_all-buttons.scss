.login-button {
    background: $blue;
    border: 2px solid $blue;
    border-radius: 5%;
    color: $black;
    display: block;
    font-size: $font-size-large; 
    font-weight: 800;
    height: 8rem;
    line-height: 1;
    margin-top: $xl-size;
    text-decoration: none;
    width: 22rem;
    transition: background .3s $royalblue;
    &:hover{
        background: $royalblue;
    }
}

.login-button_fb {
    @extend .login-button;
    background: $skyblue;
    display: block;
    color: $white;
    transition: background .3s $royalblue; 
    &:hover{
        background: $royalblue; 
    }
}

.create-button {
    @extend .login-button;
    align-content: center;
    background: $skyblue;
    font-weight: 800;
    padding-top: 3rem;
    margin: $xl-size auto;
    transition: background .3s $royalblue; 
    &:hover{
        background: $royalblue; 
    }
}

.button--link {
    background: none;
}

.save-button{
    @extend .login-button;
    background: $l-green;
    border: 2px solid $d-green;
    transition: background 0.3s $m-green;
    &:hover{
        background: $m-green;
    }
}

.remove-button {
    background: $l-gray;
    border: 2px solid $m-gray;
    margin: 2rem 0;
    transition: background .3s $d-gray;
    &:hover{
        background: $d-gray; 
    }
}