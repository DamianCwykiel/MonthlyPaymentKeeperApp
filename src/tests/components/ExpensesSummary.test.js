import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//Adapter configuration
Enzyme.configure({ adapter: new Adapter() });

test ('should corectly render expensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2355}/>);
    expect(wrapper).toMatchSnapshot();
});

test ('should corectly render expensesSummary with a few expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={20} expensesTotal={1234567890}/>);
    expect(wrapper).toMatchSnapshot();
});