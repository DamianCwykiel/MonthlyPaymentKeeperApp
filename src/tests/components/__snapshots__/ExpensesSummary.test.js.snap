// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should corectly render expensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Check your bills! There are: 
    1
     
    payment
    . In total that is: 
    23,55 zł
    .
  </h1>
</div>
`;

exports[`should corectly render expensesSummary with a few expenses 1`] = `
<div>
  <h1>
    Check your bills! There are: 
    20
     
    payments
    . In total that is: 
    12 345 678,90 zł
    .
  </h1>
</div>
`;
